src/modules/navigator/rtl.cpp

```diff
@@ -385,14 +394,16 @@ void RTL::set_rtl_item()
 			_mission_item.altitude = _rtl_alt;
 			_mission_item.altitude_is_relative = false;
 
-			// Use destination yaw if close to _destination.
-			// Check if we are pretty close to the destination already.
-			if (destination_dist < _param_rtl_min_dist.get()) {
+			if (heading_mode == 0) {
+				_mission_item.yaw = get_bearing_to_next_waypoint(gpos.lat, gpos.lon, _destination.lat, _destination.lon);
```

Comment:
Don't we still want to "Use destination yaw if close to _destination" when close to destination with the default setting?

Reply:
Good catch.  Thanks @sfuhrer, I've pushed up corrected logic.  Let me know if you have any other suggestions!